src/Cache/BaseDataCache.php

```diff
@@ -123,7 +123,7 @@ public function get_data($key, $default = null)
      * @throws InvalidArgumentException
      *   MUST be thrown if the $key string is not a legal value.
      */
-    public function set_data($key, array $value, $ttl = null)
+    public function set_data(string $key, array $value, ?int $ttl = null): bool
     {
         if (! is_int($ttl)) {
```

Comment:
This will allow us to simplify:

```suggestion
        if ($ttl === null) {
```